rust/kernel/print.rs

```diff
@@ -190,6 +203,90 @@ macro_rules! print_macro (
     );
 );
 
+/// Same as [`print_macro`], but ratelimites messages with local ratelimit_state.
+/// This emulates what [`printk_ratelimited`] macro does in C.
+///
+/// Public but hidden since it should only be used from public macros.
+///
+/// The extra parameter [`where`] is a string literal that will be used as an
+/// identifier when ratelimite is triggered. In C this is populated with
+/// __function__.
+#[doc(hidden)]
+#[cfg(not(testlib))]
+#[macro_export]
+macro_rules! print_macro_ratelimited (
+    // The non-continuation cases (most of them, e.g. `INFO`).
+    ($format_string:path, false, $where:literal, $($arg:tt)+) => ({
+        static mut PRINTK_RS: core::mem::MaybeUninit<$crate::print::ratelimit_state> =
```

Comment:
Perhaps `use` `MaybeUninit`, `AtomicPtr`, `SeqCst`, etc. to make this more readable?